src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -55,7 +86,14 @@ public JsonAdaptedPerson(Person source) {
         this.dateOfBirth = source.getDateOfBirth().toString();
         this.sex = source.getEmail().toString();
         this.status = source.getStatus().toString();
-
+        this.email = Optional.ofNullable(source.getEmail().toString());
```

Comment:
JsonAdaptedPersonTest > toModelType_validPersonDetails_returnsPerson() FAILED
    java.lang.NullPointerException at JsonAdaptedPersonTest.java:46
NullPointerException caused by calling getOptionalValues(). Instead of returning null, it caused an exception

Reply:
source.getEmail() returns a null, using toString() on a null causes the Exception.
Since Optional.ofNullable only handles the null cases within its parameters, to handle null pointer safely, we should convert it to string outside of ofNullable()